File: csharp/LogicAndTrick.WikiCodeParser/Nodes/RemovedNode.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace LogicAndTrick.WikiCodeParser.Nodes
{
    internal class RemovedNode : INode
    {
        public INode OriginalNode { get; set; }

        public RemovedNode(INode originalNode)
        {
            OriginalNode = originalNode;
        }

        public string ToHtml() => "";
        public string ToPlainText() => "";
        public IList<INode> GetChildren() => new List<INode>();
        public void ReplaceChild(int i, INode node) => throw new NotImplementedException();
        public bool HasContent() => false;
    }
}

File: php/src/Nodes/RemovedNode.php
<?php

namespace LogicAndTrick\WikiCodeParser\Nodes;

class RemovedNode implements INode
{
    public INode $originalNode;

    public function __construct(INode $originalNode)
    {
        $this->originalNode = $originalNode;
    }

    function ToHtml(): string
    {
        return '';
    }

    function ToPlainText(): string
    {
        return '';
    }

    /**
     * @inheritDoc
     */
    function GetChildren(): array
    {
        return [];
    }

    function ReplaceChild(int $i, INode $node): void
    {
        throw new \BadMethodCallException();
    }

    function HasContent(): bool
    {
        return false;
    }
}